frontend/src/components/DataTable/index.jsx

```diff
@@ -40,27 +43,52 @@ export default function DataTable({ config, DropDownRowMenu, AddNewItem }) {
 
   return (
     <>
-      <PageHeader
-        onBack={() => window.history.back()}
-        title={dataTableTitle}
-        ghost={false}
-        extra={[
-          <Button onClick={handelDataTableLoad} key={`${uniqueId()}`}>
-            Refresh
-          </Button>,
-          <AddNewItem key={`${uniqueId()}`} config={config} />,
-        ]}
-        style={{
-          padding: '20px 0px',
-        }}
-      ></PageHeader>
+      <div ref={tableHeader}>
+        <PageHeader
+          onBack={() => window.history.back()}
+          title={dataTableTitle}
+          ghost={false}
+          extra={[
+            <Button onClick={handelDataTableLoad} key={`${uniqueId()}`}>
+              Refresh
+            </Button>,
+            <AddNewItem key={`${uniqueId()}`} config={config} />,
+          ]}
+          style={{
+            padding: '20px 0px',
+          }}
+        ></PageHeader>
+      </div>
       <Table
-        columns={dataTableColumns}
+        columns={tableColumns}
         rowKey={(item) => item._id}
         dataSource={items}
         pagination={pagination}
         loading={listIsLoading}
         onChange={handelDataTableLoad}
+        expandable={
+          expandedRowData.length
+            ? {
+                expandedRowRender: (record) => (
+                  <Descriptions title="" bordered column={1}>
+                    {expandedRowData.map((item, index) => {
+                      return (
+                        <Descriptions.Item key={index} label={item.title}>
+                          {item.render?.(record[item.dataIndex])?.children
+                            ? item.render?.(record[item.dataIndex])?.children
+                            : item.render?.(record[item.dataIndex])
+                            ? item.render?.(record[item.dataIndex])
+                            : Array.isArray(item.dataIndex)
+                            ? record[item.dataIndex[0]]?.[item.dataIndex[1]]
+                            : record[item.dataIndex]}
+                        </Descriptions.Item>
```

Comment:
@salahlalami am glad you liked it 
this is because the descriptions component does not work like the table component 
and also there deferent ways that items will render
![image](https://github.com/idurar/idurar-erp-crm/assets/15654652/a2c167bb-b59d-40ae-b344-7eebb360144e)
as you can see sometimes you can just do item.dataindex 
another case : dataindex is an array
another case: there is a render that returns the value 
another case :  a render that returns a component
another case : a render returns object.children which is a component

any cleaner solution would be very appreciated